#!/usr/bin/env python

PACKAGE = "lingao_follower"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add ("h_limit_max", int_t, 0, "The maximum allowed field value Hue", 30, 0, 360)
gen.add ("h_limit_min", int_t, 0, "The minimum allowed field value Hue", 20, 0, 360)
gen.add ("s_limit_max", int_t, 0, "The maximum allowed field value Saturation", 255, 0, 256)
gen.add ("s_limit_min", int_t, 0, "The minimum allowed field value Saturation", 80, 0, 256)
gen.add ("v_limit_max", int_t, 0, "The maximum allowed field value Value", 255, 0, 256)
gen.add ("v_limit_min", int_t, 0, "The minimum allowed field value Value", 80, 0, 256)

gen.add("speed", double_t, 0, "Robot speed in meters per second", 0.2, 0.0, 1.0)
gen.add("Robot_RUN", bool_t, 0, "Check to start the robot", False)

exit(gen.generate(PACKAGE, "lingao_line_follower", "HSV_reconfigure"))
